src/harness/sas_testcase.py

```diff
@@ -165,4 +165,23 @@ def assertChannelsContainFrequencyRange(self, channels, frequency_range):
             
     channels.sort(key=lambda ch: (ch['frequencyRange']['highFrequency']), reverse = True)
     self.assertEqual(channels[0]['frequencyRange']['highFrequency'],\
-             frequency_range['highFrequency'])
\ No newline at end of file
+             frequency_range['highFrequency'])
+
+  def TriggerFullActivityDumpAndWaitUntilComplete(self, server_cert, server_key):
+    self._sas_admin.TriggerFullActivityDump()
+    request_time = datetime.utcnow()
+    signal.signal(signal.SIGALRM,
+                  lambda signum, frame:
+                  (_ for _ in ()).throw(Exception('Full Activity Dump Check Timeout')))
+    # Timeout after 2 hours if it's not completed
+    signal.alarm(7200)
+    # Check generation date of full activity dump 
+    while True:
+	  dump_message = self._sas.GetFullActivityDump( server_cert, server_key)
+	  dump_time = datetime.strptime(dump_message['generationDateTime'],
+                                               '%Y-%m-%dT%H:%M:%SZ')
+	  if request_time > dump_time:
```

Comment:
"Yes. This check should be corrected. We have made it to be <=. During testing we observed that if the fake sas return immedietly within the same second then the logic waits for one more retry. Hence we made it <= instead of <.

Also we moved the request_time = datetime.utcnow() above the self._sas_admin.TriggerFullActivityDump line."